Sources/Build/BuildDelegate.swift

```diff
@@ -237,6 +236,36 @@ public final class BuildExecutionContext {
     }
 }
 
+public protocol PackageStructureDelegate {
+    func packageStructureChanged() -> Bool
+}
+
+final class PackageStructureCommand: CustomLLBuildCommand {
+
+    override func getSignature(_ command: SPMLLBuild.Command) -> [UInt8] {
+        // Include build parameters and process env in the signature.
+        var hash = Data()
+        do {
+            hash += try self.ctx.buildParameters.hash()
+
+            let encoder = JSONEncoder()
+            if #available(macOS 10.13, *) {
+                encoder.outputFormatting = [.prettyPrinted, .sortedKeys]
+            }
+            hash += try encoder.encode(ProcessEnv.vars)
```

Comment:
I recommend `try!` on this, it can never fail, and much better than assertionFailure IMHO.